File: UI/personaltrainer_mobile/lib/models/gym.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:personaltrainer_mobile/models/image.dart';

part 'gym.g.dart';

@JsonSerializable()
class Gym {
  int? id;
  String? name;
  String? address;
  int? cityId;
  String? cityName;
  String? countryName;
  String? email;
  String? phoneNumber;
  String? workTime;
  int? imageId;
  Image? image;
  String? imageUrl;

  Gym({
    this.id,
    this.name,
    this.address,
    this.cityId,
    this.cityName,
    this.countryName,
    this.email,
    this.phoneNumber,
    this.workTime,
    this.imageId,
    this.image,
    this.imageUrl,
  });

  factory Gym.fromJson(Map<String, dynamic> json) => _$GymFromJson(json);

  Map<String, dynamic> toJson() => _$GymToJson(this);
}

File: API/eCommerce/eCommerce.Services/Database/Gym.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class Gym
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Address { get; set; } = string.Empty;

        [ForeignKey(nameof(City))]
        public int? CityId { get; set; }
        public City? City { get; set; }

        [Required]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        public string WorkTime { get; set; } = string.Empty;

        [ForeignKey(nameof(Image))]
        public int? ImageId { get; set; }
        public Image? Image { get; set; }
    }
}
